FundParser/DAL/Exceptions/NotFoundException.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FundParser.DAL.Exceptions
+{
+    public class NotFoundException : Exception
+    {
+        public NotFoundException(string? message) : base(message)
```

Comment:
Possible same changes as with other exceptions might be needed here. Also maybe `EntityNotFoundException`? Or is too long? Not Found seems a bit generic to me.